### PR TITLE
Use custom drop component

### PR DESCRIPTION
**Pull request by LukasHirt**
## Description
Remove UIKit from the drop component and used a custom one powered with popper.js

Some additional changes in this PR:
- Moved eslint config into own file
- Add `experimentalObjectRestSpread` rule
- Add medium height utility class
- Adjust drop component in breadcrumbs

### CODE REVIEW

File: src/elements/OcBreadcrumb.vue [hunk 1]
```diff
@@ -92,15 +105,44 @@ export default {
       return [...this.items].reverse()[0]
     },
   },
+
+  created() {
+    // Add a general event listener to catch outside clicks
+    document.addEventListener("click", this.documentClick)
```
Comment (LukasHirt) [inline src/elements/OcBreadcrumb.vue:111]: I'm thinking about some nice way how to do this on a more global level. A custom directive comes to my mind but am not sure about an optimal way to bring that into the target projects (e.g. ocis-web)

File: src/elements/OCDrop.vue [hunk 2]
```diff
@@ -1,155 +1,175 @@
...
+      return defaultModifiers.concat(this.modifiers)
     },
-  },
-  methods: {
-    $_ocDrop_close() {
-      if (this.closeOnClick) {
-        UiKit.drop(`#${this.dropId}`).hide()
-      }
+
+    $_ocDrop_boundary() {
```
Comment (LukasHirt) [inline src/elements/OCDrop.vue:87]: Move to computed

File: src/elements/OCDrop.vue [hunk 3]
```diff
@@ -1,155 +1,175 @@
...
+      return document.querySelector(this.boundary)
     },
+
+    $_ocDrop_element() {
```
Comment (LukasHirt) [inline src/elements/OCDrop.vue:91]: Move to computed

File: src/elements/OCDrop.vue [hunk 4]
```diff
@@ -1,155 +1,175 @@
...
+    }
   },
+
+  mounted() {
+    createPopper(this.$_ocDrop_boundary, this.$_ocDrop_element, {
+      placement: this.position,
+      modifiers: this.$_ocDrop_modifiers
+    })
+
+    if (this.matchParentWidth) {
```
Comment (LukasHirt) [inline src/elements/OCDrop.vue:102]: - [ ] React to resize of window

File: src/elements/OcBreadcrumb.vue (hunk 1 above)
Comment (LukasHirt) [reply src/elements/OcBreadcrumb.vue:111]: Register only when the dropdown is opened.

File: src/styles/theme/oc-drop.scss [hunk 5]
```diff
@@ -0,0 +1,6 @@
+.oc-drop {
+  width: 250px;
+  border: 1px solid $muted-background;
```
Comment (micbar) [inline src/styles/theme/oc-drop.scss:3]: Codacy found an issue: [Expected border to come before width (order/properties-alphabetical-order)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=5880368)

**Comment by micbar** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 5
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=5880368&bid=19247181)

File: src/styles/theme/helper.scss [hunk 6]
```diff
@@ -44,3 +44,7 @@ $directions: 'top', 'right', 'bottom', 'left';
   color: $global-inverse-color !important;
   fill: $global-inverse-color !important;
 }
+
+.oc-height-medium {
+  height: $height-medium !important;
+}
```
Comment (micbar) [inline src/styles/theme/helper.scss:50]: Codacy found an issue: [Unexpected missing end-of-source newline (no-missing-end-of-source-newline)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=5880368)

File: src/styles/theme/oc-drop.scss [hunk 7]
```diff
@@ -0,0 +1,6 @@
...
+  background-color: $background;
```
Comment (micbar) [inline src/styles/theme/oc-drop.scss:4]: Codacy found an issue: [Expected background-color to come before border (order/properties-alphabetical-order)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=5880368)

File: src/styles/_owncloud.scss [hunk 8]
```diff
@@ -47,6 +47,7 @@
 @import 'theme/oc-spinner';
 @import 'theme/oc-modal';
 @import 'theme/oc-sidebar';
+@import "theme/oc-drop";
```
Comment (micbar) [inline src/styles/_owncloud.scss:50]: Codacy found an issue: [Expected single quotes (string-quotes)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=5880368)

File: src/styles/theme/oc-drop.scss [hunk 9]
```diff
@@ -0,0 +1,6 @@
...
+  box-sizing: border-box;
+}
```
Comment (micbar) [inline src/styles/theme/oc-drop.scss:6]: Codacy found an issue: [Unexpected missing end-of-source newline (no-missing-end-of-source-newline)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=5880368)

**Comment by micbar** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=5880368&bid=19247181)

File: src/styles/theme/helper.scss (hunk 6 above)
Comment (micbar) [inline src/styles/theme/helper.scss:50]: Codacy found an issue: [Unexpected missing end-of-source newline (no-missing-end-of-source-newline)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=5880368)

**Comment by micbar** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=5880368&bid=19247181)

File: src/styles/theme/helper.scss (hunk 6 above)
Comment (micbar) [inline src/styles/theme/helper.scss:50]: Codacy found an issue: [Unexpected missing end-of-source newline (no-missing-end-of-source-newline)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=5880368)

**Comment by pascalwengerter** [comment]
@LukasHirt was this superseded by https://github.com/owncloud/owncloud-design-system/pull/1269 ? :)

**Pull request closed without merging**